pkg/cmd/pr/shared/finder.go

```diff
@@ -314,7 +318,9 @@ func findForBranch(httpClient *http.Client, repo ghrepo.Interface, baseBranch, h
 
 	for _, pr := range prs {
 		if pr.HeadLabel() == headBranch && (baseBranch == "" || pr.BaseRefName == baseBranch) {
-			return &pr, nil
+			if pr.State == "OPEN" || resp.Repository.DefaultBranchRef.Name != headBranch {
```

Comment:
This condition looks good but could be merged with the `if` condition that precedes it. Something like:

```go
if pr.HeadLabel() == headBranch && (baseBranch == "" || pr.BaseRefName == baseBranch) && (pr.State == "OPEN" || resp.Repository.DefaultBranchRef.Name != headBranch) {
```

Feel free to split the condition into multiple lines if that looks better

Reply:
I am of 2 minds 😅. I can do it however you like.

Reply:
Well then just put it on 1-line `if` condition and then we'll see. As long as it's not nested `if`s 👍

Reply:
Done